pyop2/types/dataset.py

```diff
@@ -244,7 +251,7 @@ def set(self):
     @utils.cached_property
     def size(self):
         """The number of local entries in the Dataset (1 on rank 0)"""
-        return 1 if mpi.MPI.comm.rank == 0 else 0
+        return 1 if self.comm.rank == 0 else 0
```

Comment:
This should be put in a distinct commit as it is unrelated to the rest of the changes here.